packages/plugin/src/rules/strict-id-in-types.ts

```diff
@@ -0,0 +1,172 @@
+      requiresSiblings: false,
+      url: 'https://github.com/dotansimha/graphql-eslint/blob/master/docs/rules/strict-id-in-types.md',
+      examples: [
+        {
+          title: 'Incorrect',
+          usage: [{ acceptedIdNames: ['id', '_id'], acceptedIdTypes: ['ID'], exceptions: { suffixes: ['Payload'] } }],
+          code: /* GraphQL */ `
+            # Incorrect field name
+            type InvalidFieldName {
+              key: ID!
+            }
+
+            # Incorrect field type
+            type InvalidFieldType {
+              id: String!
+            }
+
+            # Incorrect exception suffix
+            type InvalidSuffixResult {
+              data: String!
+            }
+
+            # Too many unique identifiers. Must only contain one.
+            type InvalidFieldName {
+              id: ID!
+              _id: ID!
+            }
+          `,
+        },
+        {
+          title: 'Correct',
+          usage: [{ acceptedIdNames: ['id', '_id'], acceptedIdTypes: ['ID'], exceptions: { suffixes: ['Payload'] } }],
+          code: /* GraphQL */ `
+            type User {
+              id: ID!
+            }
+
+            type Post {
+              _id: ID!
+            }
+
+            type CreateUserPayload {
+              data: String!
+            }
+          `,
+        },
+      ],
+    },
+    schema: {
+      $schema: 'http://json-schema.org/draft-04/schema#',
+      type: 'array',
+      items: {
+        type: 'object',
+        properties: {
+          acceptedIdNames: {
+            type: 'array',
+            items: {
+              type: 'string',
+            },
+            default: ['id'],
+          },
+          acceptedIdTypes: {
+            type: 'array',
+            items: {
+              type: 'string',
+            },
+            default: ['ID'],
+          },
+          exceptions: {
+            type: 'object',
+            properties: {
+              suffixes: {
+                type: 'array',
+                items: {
+                  type: 'string',
+                },
+                default: [],
+              },
+            },
+          },
+        },
+      },
+    },
+  },
+  create(context) {
+    const options: StrictIdInTypesRuleConfig[number] = {
+      acceptedIdNames: ['id'],
+      acceptedIdTypes: ['ID'],
+      exceptions: {},
+      ...(context.options[0] || {}),
+    };
+
+    return {
+      ObjectTypeDefinition(node) {
+        if (shouldIgnoreNode({ node, exceptions: options.exceptions })) {
+          return;
+        }
+
+        const validIds = node.fields.filter(field => {
+          const fieldNode = field.rawNode();
+
+          const isValidIdName = options.acceptedIdNames.includes(fieldNode.name.value);
+
+          // To be a valid type, it must be non-null and one of the accepted types.
+          let isValidIdType = false;
+          if (fieldNode.type.kind === 'NonNullType' && fieldNode.type.type.kind === 'NamedType') {
```

Comment:
Is there a better way to check node types using some kind of helper from the `graphql` package? I tried so many approaches but couldn't get this working so I went with this. :)

I'm hoping to get some advice/guidance here 🙏

Reply:
`graphql.js` has `isNonNullType` but it's not for AST, it's only for class types. 

I'm totally fine with the way you tested it, but maybe you can use `Kind.NON_NULL_TYPE` and `Kind.NAMED_TYPE` instead of the explicit string valus?

Reply:
Makes sense! This is done in https://github.com/dotansimha/graphql-eslint/pull/333/commits/6ba418a98de2823f98a605e117f034fb12090d39. Cheers!